src/main/java/io/mosip/digitalcard/service/impl/PrintServiceImpl.java

```diff
@@ -0,0 +1,603 @@
+		List<String> list = new ArrayList<>(Arrays.asList(attributes));
+
+		Iterator<String> it = list.iterator();
+		String uinCardPd = "";
+		Object obj=null;
+		while (it.hasNext()) {
+			String key = it.next().trim();
+
+			Object object = jsonObject.get(key);
+			if (object != null) {
+				try {
+					obj = new JSONParser().parse(object.toString());
+				} catch (Exception e) {
+					obj = object;
+				}
+			}
+			if (obj instanceof JSONArray) {
+				// JSONArray node = JsonUtil.getJSONArray(demographicIdentity, value);
+				JsonValue[] jsonValues = Utility.mapJsonNodeToJavaObject(JsonValue.class, (JSONArray) obj);
+				uinCardPd = uinCardPd.concat(getParameter(jsonValues, templateLang).substring(0,4));
+			} else if (object instanceof JSONObject) {
+				JSONObject json = (JSONObject) object;
+				uinCardPd = uinCardPd.concat((String) json.get(VALUE));
+			} else {
+				uinCardPd = uinCardPd.concat((String) object.toString().substring(0,4));
+			}
+		}
+		return uinCardPd;
+	}
+
+	/**
+	 * Gets the parameter.
+	 *
+	 * @param jsonValues
+	 *            the json values
+	 * @param langCode
+	 *            the lang code
+	 * @return the parameter
+	 */
+	private String getParameter(JsonValue[] jsonValues, String langCode) {
+
+		String parameter = null;
+		if (jsonValues != null) {
+			for (int count = 0; count < jsonValues.length; count++) {
+				String lang = jsonValues[count].getLanguage();
+				if (langCode.contains(lang)) {
+					parameter = jsonValues[count].getValue();
+					break;
+				}
+			}
+		}
+		return parameter;
+	}
+
+
+
+	private String getCrdentialSubject(String crdential) throws JSONException {
+		org.json.JSONObject jsonObject = new org.json.JSONObject(crdential);
+		String credentialSubject = jsonObject.get("credentialSubject").toString();
+		return credentialSubject;
+	}
+
+	private void printStatusUpdate(String requestId, byte[] data, String credentialType, String rid)
+			throws DataShareException, ApiNotAccessibleException, IOException, Exception {
+		DataShare dataShare = null;
+		dataShare = dataShareUtil.getDataShare(data, policyId, partnerId);
+		CredentialStatusEvent creEvent = new CredentialStatusEvent();
+		LocalDateTime currentDtime = DateUtils.getUTCCurrentDateTime();
+		digitalCardTransactionRepository.updateTransactionDetails(rid,"AVAILABLE",dataShare.getUrl());
+		StatusEvent sEvent = new StatusEvent();
+		sEvent.setId(UUID.randomUUID().toString());
+		sEvent.setRequestId(requestId);
+		sEvent.setStatus("STORED");
+		sEvent.setUrl(dataShare.getUrl());
+		sEvent.setTimestamp(Timestamp.valueOf(currentDtime).toString());
+		creEvent.setPublishedOn(LocalDateTime.now().toString());
+		creEvent.setPublisher("PRINT_SERVICE");
+		creEvent.setTopic(topic);
+		creEvent.setEvent(sEvent);
+		webSubSubscriptionHelper.printStatusUpdateEvent(topic, creEvent);
+	}
+/*
+	public org.json.JSONObject decryptAttribute(org.json.JSONObject data, String encryptionPin, String credential)
```

Comment:
If this is not required, lets remove it.

Reply:
taken care